docs/content/config/security/ssl.md

```diff
@@ -427,39 +427,68 @@ This setup only comes with one caveat: The domain has to be configured on anothe
 
     Use self-signed certificates only for testing purposes!
 
-You can  generate a self-signed SSL certificate by using the following command:
+This feature requires you to provide the following files into your [`config/ssl/` directory][docs-optional-config] (internal location: `/tmp/docker-mailserver/ssl/`):
 
-```sh
-docker run -it --rm -v "$(pwd)"/config/ssl:/tmp/docker-mailserver/ssl -h mail.my-domain.com -t mailserver/docker-mailserver generate-ssl-certificate
-
-# Press enter
-# Enter a password when needed
-# Fill information like Country, Organisation name
-# Fill "my-domain.com" as FQDN for CA, and "mail.my-domain.com" for the certificate.
-# They HAVE to be different, otherwise you'll get a `TXT_DB error number 2`
-# Don't fill extras
-# Enter same password when needed
-# Sign the certificate? [y/n]:y
-# 1 out of 1 certificate requests certified, commit? [y/n]y
-
-# will generate:
-# config/ssl/mail.my-domain.com-key.pem (used in postfix)
-# config/ssl/mail.my-domain.com-req.pem (only used to generate other files)
-# config/ssl/mail.my-domain.com-cert.pem (used in postfix)
-# config/ssl/mail.my-domain.com-combined.pem (used in courier)
-# config/ssl/demoCA/cacert.pem (certificate authority)
-```
+- `${HOSTNAME}-key.pem`
+- `${HOSTNAME}-cert.pem`
+- `demoCA/cacert.pem`
 
-!!! note
-    The certificate will be generate for the container `fqdn`, that is passed as `-h` argument.
-
-    Check the following page for more information regarding [postfix and SSL/TLS configuration](http://www.mad-hacking.net/documentation/linux/applications/mail/using-ssl-tls-postfix-courier.xml).
+Where `${HOSTNAME}` is the mailserver [FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name) (`hostname`(_mail_) + `domainname`(_example.com_), eg: `mail.example.com`).
 
 To use the certificate:
 
-- Add `SSL_TYPE=self-signed` to your container environment variables
+- Add `SSL_TYPE=self-signed` to your container environment variables.
 - If a matching certificate (files listed above) is found in `config/ssl`, it will be automatically setup in postfix and dovecot. You just have to place them in `config/ssl` folder.
 
+#### Generating a self-signed certificate
+
+!!! note
+
+    Since v10, support in `setup.sh` for generating a self-signed SSL certificate internally was removed.
+    
+    It is now similar to `SSL_TYPE=manual` (_except `manual` does not support verification for a custom CA_), but does not require additional ENV vars for providing the location of cert files.
+
+One way to generate self-signed certificates is with [Smallstep's `step` CLI](https://smallstep.com/docs/step-cli). This is exactly what `docker-mailserver` does for creating test certificates](https://github.com/docker-mailserver/docker-mailserver/tree/master/test/test-files/ssl/example.test).
```

Comment:
I probably shouldn't have been lazy with this contribution and previewed docs locally :sweat_smile: 

```suggestion
One way to generate self-signed certificates is with [Smallstep's `step` CLI](https://smallstep.com/docs/step-cli). This is exactly what [`docker-mailserver` does for creating test certificates](https://github.com/docker-mailserver/docker-mailserver/tree/master/test/test-files/ssl/example.test).
```

This should technically be a `[github-*]` link added at the bottom of the docs too..